### PR TITLE
feat: standardize event data for mint and burn none-interest

### LINKED ISSUE #285: Standarize Event Data for mint and burn none-interest
1. Read our [Contributor Guidelines](https://github.com/CarmineOptions/derisk-research/blob/master/CONTRIBUTING.md) and [README](https://github.com/CarmineOptions/derisk-research/blob/master/README.md). 
2. reference how it was done fore `Zklend` you can find it [here](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handler_tools/data_parser/zklend.py#L17)
3. In `nostra.py`  in class [NostraDataParser](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handler_tools/data_parser/nostra.py#L4) finish next methods:
- [parse_non_interest_bearing_collateral_mint_event](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handler_tools/data_parser/nostra.py#L11C9-L11C57)
- [parse_non_interest_bearing_collateral_burn_event](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handler_tools/data_parser/nostra.py#L14C9-L14C57)
4. Your task is Create serializer for each even data for those methods which you can find here:
- [process_non_interest_bearing_collateral_mint_event](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handlers/loan_states/nostra_alpha/events.py#L359C9-L359C59)
- [process_non_interest_bearing_collateral_burn_event](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handlers/loan_states/nostra_alpha/events.py#L492C9-L492C59)
5. Don't forget to update `process_non_interest_bearing_collateral_mint_event` and `process_non_interest_bearing_collateral_burn_event` using your serializing models

### PR DESCRIPTION
**Pull request by Tzienom**
Closes #285

### CODE REVIEW

File: apps/data_handler/handlers/loan_states/nostra_alpha/events.py [hunk 1]
```diff
@@ -372,9 +372,10 @@ def process_non_interest_bearing_collateral_mint_event(self, event: pd.Series) -
             sender = add_leading_zeros(event["data"][0])
             recipient = add_leading_zeros(event["data"][1])
             raw_amount = decimal.Decimal(str(int(event["data"][2], base=16)))
+            parsed_event = NostraDataParser.parse_non_interest_bearing_collateral_mint_event(event["data"])
         else:
             raise ValueError("Event = {} has an unexpected structure.".format(event))
-        if self.ZERO_ADDRESS in {sender, recipient}:
+        if self.ZERO_ADDRESS in {sender, recipient} | self.ZERO_ADDRESS in {parsed_event.sender, parsed_event.recipient}:
```
Comment (djeck1432) [inline apps/data_handler/handlers/loan_states/nostra_alpha/events.py:378]: why diid you change this code?

File: apps/data_handler/handlers/loan_states/nostra_alpha/events.py [hunk 2]
```diff
@@ -398,6 +399,26 @@ def process_non_interest_bearing_collateral_mint_event(self, event: pd.Series) -
                 )
             )
 
+        if parsed_event.sender != self.DEFERRED_BATCH_CALL_ADAPTER_ADDRESS:
```
Comment (djeck1432) [inline apps/data_handler/handlers/loan_states/nostra_alpha/events.py:402]: You have duplicated code, please remove it

**Comment by djeck1432** [comment]
@Tzienom I'm not sure that you can manage with this task, so easy task and so many issues

**Comment by Tzienom** [comment]
i will fix it before end of day

File: apps/data_handler/handlers/loan_states/nostra_alpha/events.py [hunk 3]
```diff
@@ -372,6 +372,7 @@ def process_non_interest_bearing_collateral_mint_event(self, event: pd.Series) -
             sender = add_leading_zeros(event["data"][0])
```
Comment (djeck1432) [inline apps/data_handler/handlers/loan_states/nostra_alpha/events.py:372]: why you didn't remove these fields?

File: apps/data_handler/handlers/loan_states/nostra_alpha/events.py [hunk 4]
```diff
@@ -500,26 +502,40 @@ def process_non_interest_bearing_collateral_burn_event(self, event: pd.Series) -
         # Example:
         # https://starkscan.co/event/0x00744177ee88dd3d96dda1784e2dff50f0c989b7fd48755bc42972af2e951dd6_1.
         user = event["data"][0]
-        if user == self.IGNORE_USER:
+        parsed_event = NostraDataParser.parse_non_interest_bearing_collateral_burn_event(event["data"])
+        if user == self.IGNORE_USER | parsed_event == self.IGNORE_USER:
```
Comment (djeck1432) [inline apps/data_handler/handlers/loan_states/nostra_alpha/events.py:506]: why did you change this code?
and why parsed event return only user data here if it should return more values?

File: apps/data_handler/handlers/loan_states/nostra_alpha/events.py [hunk 5]
```diff
@@ -500,26 +502,40 @@ def process_non_interest_bearing_collateral_burn_event(self, event: pd.Series) -
...
         }
-        if user == self.verbose_user:
+        self.loan_entities[parsed_event.user].collateral.values = {
+            token: (
+                self.loan_entities[parsed_event.user].non_interest_bearing_collateral.values[token] +
+                self.loan_entities[parsed_event.user].interest_bearing_collateral.values[token]
+            )
+            for token in NOSTRA_ALPHA_SPECIFIC_TOKEN_SETTINGS
+        }
+        if user == self.verbose_user | parsed_event.user == self.verbose_user:
```
Comment (djeck1432) [inline apps/data_handler/handlers/loan_states/nostra_alpha/events.py:538]: why did you change it

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Okay, last attempt, please, do not override main logic, you just need to substitute parsing event["data"] with your parser and use these fields

**Comment by djeck1432** [comment]
@Tzienom Please fix your comments within next hours, otherwise you will be unassigned, this task blocking us to go further

**Comment by Tzienom** [comment]
completing shortly

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Please, sync with master

**Comment by Tzienom** [comment]
@djeck1432 my branch is currently uptodate with master already

**Pull request closed without merging**